pkgs/development/tools/apko/default.nix

```diff
@@ -0,0 +1,72 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, installShellFiles
+}:
+
+buildGoModule rec {
+  pname = "apko";
+  version = "0.3.3";
+
+  src = fetchFromGitHub {
+    owner = "chainguard-dev";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-j/tfa9O10HSEwzuXE2mjV2TVYuwk9lpA0PgsKDm70uI=";
+    # populate values that require us to use git. By doing this in postFetch we
+    # can delete .git afterwards and maintain better reproducibility of the src.
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git rev-parse HEAD > $out/COMMIT
+      # '0000-00-00T00:00:00Z'
+      date -u -d "@$(git log -1 --pretty=%ct)" "+'%Y-%m-%dT%H:%M:%SZ'" > $out/SOURCE_DATE_EPOCH
+      find "$out" -name .git -print0 | xargs -0 rm -rf
+    '';
```

Comment:
Can we just use ``src.rev`` instead of this? Its way easier, has less complexity and can't break reproducibility that easily.

Reply:
Hmmm, I still don't like it that way. We also cannot easily get it some other way from the fetcher... 🤔

Reply:
I thought about modifying the fetcher to have a `withMetadata` flag or something and that would go in a separate output or something, I need to play with it

My first thought around it was it would also be this kind of fetchgit then get the details but then I thought maybe it could grab the tarball and just fetch the other details via APIs etc

Reply:
We could need to do API calls only on demand or when credentials are supplied which makes this rather hard.

Reply:
Good point